test/unit/cwe_checker_unit_tests.ml

```diff
@@ -1,26 +1,89 @@
 open Bap.Std
 open Core_kernel
 
+include Self()
+
+let cmdline_params = [
+  ("partial", "Comma separated list defining which tests should be executed with the current test file. e.g. MemRegion,TypeInference,CWE476,...")
+]
+
+
+let input_test_map = Map.of_alist_exn (module String) [
+  "MemRegion", "Mem_region_tests";
+  "TypeInference", "Type_inference_tests";
+  "CWE476", "CWE_476_tests";
+  "CWE560", "CWE_560_tests";
+  "AddrTrans", "Address_translation_tests";
+  "Cconv", "Cconv_tests"
+]
+
+let unit_test_list = [
+  "Mem_region_tests", Mem_region_test.tests;
+  "Type_inference_tests", Type_inference_test.tests;
+  "Cconv_tests", Cconv_test.tests;
+  "CWE_476_tests", Cwe_476_test.tests;
+  "CWE_560_tests", Cwe_560_test.tests;
+  "Address_translation_tests", Address_translation_test.tests;
+]
+
+
+let set_example_project (project : Project.t) (tests : string list) =
+  List.iter tests ~f:(fun test ->
+    match test with
+    | "Type_inference_tests" -> Type_inference_test.example_project := Some(project)
+    | "Cconv_tests" -> Cconv_test.example_project := Some(project)
+    | "CWE_476_tests" -> Cwe_476_test.example_project := Some(project)
+    | _ -> ()
+  )
+
+
+let full_run (project : Project.t) =
+  set_example_project project (List.map unit_test_list ~f:(fun test -> match test with (key, _) -> key));
+  Alcotest.run "Unit tests" ~argv:[|"DoNotComplainWhenRunAsABapPlugin";"--color=always";|] unit_test_list
```

Comment:
If the tests expect different binaries as input files, is a full run actually possible? If not, we should remove the option for it.